wcfsetup/install/files/lib/system/background/job/UnfurlUrlBackgroundJob.class.php

```diff
@@ -0,0 +1,243 @@
+<?php
+
+namespace wcf\system\background\job;
+
+use BadMethodCallException;
+use GuzzleHttp\Psr7\Response;
+use wcf\data\unfurl\url\UnfurlUrl;
+use wcf\data\unfurl\url\UnfurlUrlAction;
+use wcf\system\message\unfurl\exception\DownloadFailed;
+use wcf\system\message\unfurl\exception\ParsingFailed;
+use wcf\system\message\unfurl\exception\UrlInaccessible;
+use wcf\system\message\unfurl\UnfurlResponse;
+use wcf\util\FileUtil;
+use wcf\util\StringUtil;
+
+/**
+ * Represents a background job to get information for an url.
+ *
+ * @author      Joshua Ruesweg
+ * @copyright   2001-2020 WoltLab GmbH
+ * @license     GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package     WoltLabSuite\Core\System\Background\Job
+ * @since       5.4
+ */
+final class UnfurlUrlBackgroundJob extends AbstractBackgroundJob
+{
+    /**
+     * @var int
+     */
+    private $urlID;
+
+    /**
+     * UnfurlURLJob constructor.
+     *
+     * @param UnfurlUrl $url
+     */
+    public function __construct(UnfurlUrl $url)
+    {
+        $this->urlID = $url->urlID;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function retryAfter()
+    {
+        switch ($this->getFailures()) {
+            case 1:
+                // 5 minutes
+                return 5 * 60;
+            case 2:
+                // 30 minutes
+                return 30 * 60;
+            case 3:
+                // 2 hours
+                return 2 * 60 * 60;
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function perform()
+    {
+        $unfurlUrl = new UnfurlUrl($this->urlID);
+
+        try {
+            $unfurlResponse = UnfurlResponse::fetch($unfurlUrl->url);
+
+            if (empty(StringUtil::trim($unfurlResponse->getTitle()))) {
+                $this->save(UnfurlUrl::STATUS_REJECTED);
+
+                return;
+            }
+
+            $title = StringUtil::truncate($unfurlResponse->getTitle(), 255);
+            $description = "";
+            if ($unfurlResponse->getDescription()) {
+                $description = StringUtil::truncate($unfurlResponse->getDescription());
```

Comment:
```suggestion
                $description = StringUtil::truncate($unfurlResponse->getDescription(), 1000);
```
Otherwise the method will default to 80.